contracts/HubPool.sol

```diff
@@ -72,4 +125,29 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         uint256[] memory netSendAmounts,
         bytes32[] memory inclusionProof
     ) public {}
+
+    function _exchangeRateCurrent() internal pure returns (uint256) {
+        return 1e18;
+    }
+
+    // Unwraps ETH and does a transfer to a recipient address. If the recipient is a smart contract then sends WETH.
+    function _unwrapWETHTo(address payable to, uint256 amount) internal {
+        if (address(to).isContract()) {
+            IERC20(address(l1Weth)).safeTransfer(to, amount);
+        } else {
+            l1Weth.withdraw(amount);
+            to.transfer(amount);
+        }
+    }
+
+    function append(
+        string memory a,
+        string memory b,
+        string memory c
+    ) internal pure returns (string memory) {
+        return string(abi.encodePacked(a, b, c));
+    }
+
+    // Added to enable the BridgePool to receive ETH. used when unwrapping Weth.
+    receive() external payable {}
```

Comment:
Should we revert if this comes from an address that is not the WETH contract?

Reply:
that's an interesting idea but its somewhat inconsistant. we dont prevent users from sending tokens so why should we prevent them from sending eth?

Reply:
I see what you're saying. ETH transfers work differently, so I would argue it's more likely to send ETH accidentally. However, this would incur a slight gas cost, so it probably isn't worth the cost of the check.